src/veh_interact.h

```diff
@@ -150,8 +150,13 @@ class veh_interact
         std::string totalDurabilityText;
         nc_color totalDurabilityColor;
 
-        /** Store the most damaged part's index, or -1 if they're all healthy. */
-        int mostDamagedPart = -1;
+        /** Returns the most damaged part's index, or -1 if they're all healthy. */
+        int get_most_damaged_part() const;
+
+        /** Returns the index of the part that needs repair the most.
+         * This may not be mostDamagedPart since not all parts can be repaired
+         * If there are no damaged parts this returns -1 */
+        int get_most_repariable_part() const;
```

Comment:
Yet another improvement could be making these functions return pointers:
```c+
const vehicle_part *get_most_repariable_part() const;
```
This way you can:
- Get rid of the magic numbers (`-1`). Comparison with `nullptr` is clearer.
- Replace the cumbersome `veh->parts[most_repariable].whatever` with the nicer `most_repariable->whatever`.